mod-source-record-manager-server/src/main/java/org/folio/services/util/EventHandlingUtil.java

```diff
@@ -39,45 +42,50 @@ private EventHandlingUtil() {
-      .withJobExecutionId(record.getSnapshotId())
-      .withContext(dataImportEventPayloadContext)
-      .withOkapiUrl(params.getOkapiUrl())
-      .withTenant(params.getTenantId())
-      .withToken(params.getToken());
+      DataImportEventPayload dataImportEventPayload = new DataImportEventPayload()
+        .withEventType(eventType)
+        .withProfileSnapshot(profileSnapshotWrapper)
+        .withCurrentNode(profileSnapshotWrapper.getChildSnapshotWrappers().get(0))
+        .withJobExecutionId(record.getSnapshotId())
+        .withContext(dataImportEventPayloadContext)
+        .withOkapiUrl(params.getOkapiUrl())
+        .withTenant(params.getTenantId())
+        .withToken(params.getToken());
 
-    Event createdRecordEvent = new Event()
-      .withId(UUID.randomUUID().toString())
-      .withEventType(eventType)
-      .withEventPayload(Json.encode(dataImportEventPayload))
-      .withEventMetadata(new EventMetadata()
-        .withTenantId(params.getTenantId())
-        .withEventTTL(1)
-        .withPublishedBy(PubSubClientUtils.constructModuleName()));
+      Event createdRecordEvent = new Event()
+        .withId(UUID.randomUUID().toString())
+        .withEventType(eventType)
+        .withEventPayload(ZIPArchiver.zip(Json.encode(dataImportEventPayload)))
+        .withEventMetadata(new EventMetadata()
+          .withTenantId(params.getTenantId())
+          .withEventTTL(1)
+          .withPublishedBy(PubSubClientUtils.constructModuleName()));
 
-    org.folio.rest.util.OkapiConnectionParams connectionParams = new org.folio.rest.util.OkapiConnectionParams();
-    connectionParams.setOkapiUrl(params.getOkapiUrl());
-    connectionParams.setToken(params.getToken());
-    connectionParams.setTenantId(params.getTenantId());
-    connectionParams.setVertx(params.getVertx());
+      org.folio.rest.util.OkapiConnectionParams connectionParams = new org.folio.rest.util.OkapiConnectionParams();
+      connectionParams.setOkapiUrl(params.getOkapiUrl());
+      connectionParams.setToken(params.getToken());
+      connectionParams.setTenantId(params.getTenantId());
+      connectionParams.setVertx(params.getVertx());
 
-    PubSubClientUtils.sendEventMessage(createdRecordEvent, connectionParams)
-      .whenComplete((ar, throwable) -> {
-        if (throwable == null) {
-          promise.complete(record);
-        } else {
-          LOGGER.error("Error during event sending: {}", throwable, createdRecordEvent);
-          promise.fail(throwable);
-        }
-      });
+      PubSubClientUtils.sendEventMessage(createdRecordEvent, connectionParams)
+        .whenComplete((ar, throwable) -> {
+          if (throwable == null) {
+            promise.complete(record);
+          } else {
+            LOGGER.error("Error during event sending: {}", throwable, createdRecordEvent);
+            promise.fail(throwable);
+          }
+        });
+    } catch (Exception e) {
+      LOGGER.error("Error during event building or sending: {}", e);
```

Comment:
```suggestion
      LOGGER.error("Error during event building or sending", e);
```
no need to log exception as a param

Reply:
Sure, thanks